tests/test_image_driver.py

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env python 
+''' VIC Image Driver testing '''
+
+import xarray as xr
+import numpy as np 
+import numpy.testing as npt 
+
+def assert_nan_equal(ds_domain, ds_output): 
+    """ 
+    test to see if nans in two data arrays are the same in order to check domain 
+    first dataarray is domain file, second dataarray is a variable in the output
+    dataset for image driver 
+    """
+
+    # get lat/lng dim names from domain file 
+    lng_name, lat_name = ds_domain.coords
+
+    match_vars = [lng_name, lat_name] 
+    
+    # check to be sure that mask, lats and lons match between domain file and output file
+    for var in match_vars: 
+        # raise AssertionError if they don't match
+        npt.assert_allclose(ds_output[var], ds_domain[var], equal_nan=True) 
+
+    # check that nans are occurring in the same place in the arrays 
+    
+    # check all variables in the dataset 
+    for da in ds_output.data_vars: 
+        
+        # check all timesteps 
+        for ts in range(0, len(ds_output.time)):
```

Comment:
Nothing outright wrong here but the for loop isn't required. I still think using `isull` makes more since here. Some pseudo code for you:

```Python
# Find grid cells that are never nans (that mask should match the domain mask)
da_notnull = da.notnull().all(dim=('time', ..., other_non_space_dims))
# compare the masks
npt.assert_array_equal(da.values, ds_domain['mask'].values)
```